site/public/js/server.js

```diff
@@ -870,6 +870,35 @@ function markViewed(ids, redirect) {
     });
 }
 
+function deleteBadges() {
+    // Delete badges
+    var badges = document.querySelectorAll('.course-material-badge');
+    badges.forEach(function(badge) {
+        badge.remove();
+    });
+    let data = new FormData();
+    data.append("csrf_token", csrfToken);
+    $.ajax({
+        url: buildCourseUrl(['course_materials', 'viewAll']), 
+        type: "POST",
+        data: data,
+        contentType: false,
+        processData: false,
+        success: function(response) {
+            console.log("Success:", response);
+            // Optional: Redirect or perform some action on success
+            if (redirect) {
+                window.location.href = redirect;
```

Comment:
Why are you executing an AJAX request and then promptly reloading the page?  You could handle the redirect on the server side.